src/bgw/job.c

```diff
@@ -1317,18 +1318,18 @@ ts_bgw_job_run_and_set_next_start(BgwJob *job, job_main_func func, int64 initial
 								  Interval *next_interval, bool atomic, bool mark)
 {
 	BgwJobStat *job_stat;
-	bool had_error;
+	JobResult result;
 
 	if (atomic)
 		StartTransactionCommand();
 
 	if (mark)
 		ts_bgw_job_stat_mark_start(job->fd.id);
 
-	had_error = func();
+	result = func();
```

Comment:
`job_main_func` returns `bool` and not `JobResult`

Reply:
Switched to JobResult.